packages/block-library/src/button/edit.native.js

```diff
@@ -339,273 +332,233 @@ class ButtonEdit extends Component {
 		}
 	}
 
-	dismissSheet() {
-		this.onHideLinkSettings();
-		this.props.closeSettingsBottomSheet();
+	function onPlaceholderTextWidth( { nativeEvent } ) {
+		const textWidth =
+			nativeEvent.lines[ 0 ] && nativeEvent.lines[ 0 ].width;
+
+		if ( textWidth && textWidth !== placeholderTextWidth ) {
+			setPlaceholderTextWidth( Math.min( textWidth, maxWidth ) );
+		}
 	}
 
-	getLinkSettings( isCompatibleWithSettings ) {
-		const { isLinkSheetVisible } = this.state;
-		const { attributes, setAttributes } = this.props;
+	const onSetRef = useCallback(
+		( ref ) => {
+			richTextRef.current = ref;
+		},
+		[ richTextRef ]
+	);
+
+	const onUnstableOnFocus = useCallback( () => {
+		onToggleButtonFocus( true );
+	}, [ onToggleButtonFocus ] );
+
+	const onBlur = useCallback( () => {
+		onSetMaxWidth();
+	}, [ onSetMaxWidth ] );
```

Comment:
Do you know what difference does it make to pass check for the onSetMaxWidth each time? 

`[ onSetMaxWidth ]` can we just remove that argument?